common/aws/provider/connection.go

```diff
@@ -75,7 +79,24 @@ var getConfig = func(credProvider CredProvider, region string) (sess *session.Se
 		Credentials: creds,
 		Region:      aws.String(region),
 	}
+
+	maxRetries, err := strconv.Atoi(config.AWSMaxRetries())
```

Comment:
in theory, this logic should only need to exist in 1 location; I'm seeing a lot of copies everywhere.